### PR TITLE
Let the admin convert a forum into a private room

### PR DESCRIPTION
**Pull request by aulamber**
Cf. issue https://github.com/tchapgouv/tchap-web-v4/issues/363

To reproduce : 
1. Create a forum (you need to be an admin of your forum).
2. Go to "Paramètres du salon" > "Général".
3. A new button appears : "Convertir en salon privé". Click on it, it will change your public room into a private one.

Capture d’écran 2023-06-09 à 17 10 54

NB : I had trouble updating the list of rooms (outside of our component) so that the green forum icon of our newly turned private room would turn into a red lock icon. It is only a frontend problem, because I added a request to update the room status and it shows when I refresh the page. So my solution was to refresh the page with javascript. If you know of a better solution in the existing matrix code, please let me know :) In this component: 

Capture d’écran 2023-06-09 à 16 54 07

NB : the way I check that the user is an admin of the room is by verifying their powerLevel (100 as an admin, 0 as a regular user).

### CODE REVIEW

File: patches/activate-cross-signing-and-secure-storage-js/matrix-js-sdk+25.0.0.patch [hunk 1]
```diff
@@ -20,7 +20,76 @@ index cb8df7a..1ca2f21 100644
  export type Store = IStore;
  
  export type ResetTimelineCallback = (roomId: string) => boolean;
-@@ -7284,6 +7288,17 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
+@@ -4252,6 +4256,60 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
+         return this.http.authedRequest(Method.Put, path, undefined, content);
+     }
+ 
++    /**
```
Comment (odelcroi) [inline patches/activate-cross-signing-and-secure-storage-js/matrix-js-sdk+25.0.0.patch]: can it be outside of the matrix-js-sdk?

File: patches/hide-room-alias-settings/matrix-react-sdk+3.71.1.patch [hunk 2]
```diff
@@ -1,20 +1,127 @@
...
++            changePublicToPrivateRoomLoader: false, // :TCHAP:
++            isRoomPublic: joinRules?.getContent()["join_rule"] === JoinRule.Public ?? true, // :TCHAP:
+             isRoomPublished: false, // loaded async
+         };
+     }
+@@ -56,6 +70,35 @@ export default class GeneralRoomSettingsTab extends React.Component<IProps, ISta
+         PosthogTrackers.trackInteraction("WebRoomSettingsLeaveButton", ev);
+     };
+ 
++    /* :TCHAP: Add button to change forum to private room if user is admin */
```
Comment (odelcroi) [inline patches/hide-room-alias-settings/matrix-react-sdk+3.71.1.patch]: can it be outside of the matrix-react-sdk

**Comment by estellecomment** [comment]
C'est tjs d'actualité ca ?

File: patches/activate-cross-signing-and-secure-storage-js/matrix-js-sdk+25.0.0.patch [hunk 3]
```diff
@@ -20,7 +20,15 @@ index cb8df7a..1ca2f21 100644
  export type Store = IStore;
  
  export type ResetTimelineCallback = (roomId: string) => boolean;
-@@ -7284,6 +7288,17 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
+@@ -5951,6 +5955,7 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
+         if (filter) {
+             params.filter = JSON.stringify(filter);
+         }
++
```
Comment (odelcroi) [inline patches/activate-cross-signing-and-secure-storage-js/matrix-js-sdk+25.0.0.patch:27]: unecessary add line break

**Comment by odelcroi** [comment]
We need to double check the implementation and add tests BUT it is not a huge priority as this use case arose only once.

**Pull request closed without merging**